cmd/ci-operator-configresolver/main.go

```diff
@@ -119,6 +119,99 @@ func resolveConfig(configAgent agents.ConfigAgent, registryAgent agents.Registry
 	}
 }
 
+const (
+	injectFromOrgQuery     = "injectTestFromOrg"
+	injectFromRepoQuery    = "injectTestFromRepo"
+	injectFromBranchQuery  = "injectTestFromBranch"
+	injectFromVariantQuery = "injectTestFromVariant"
+	injectTestQuery        = "injectTest"
+)
+
+func injectTestFromQuery(w http.ResponseWriter, r *http.Request) (api.Metadata, string, error) {
+	var metadata api.Metadata
+	var test string
+
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusNotImplemented)
+		err := fmt.Errorf("expected GET, got %s", r.Method)
+		if _, errWrite := w.Write([]byte(http.StatusText(http.StatusNotImplemented))); errWrite != nil {
+			err = fmt.Errorf("%s and writing the response body failed with %w", err.Error(), errWrite)
+		}
+		return metadata, test, err
+	}
+	metadata.Org = r.URL.Query().Get(injectFromOrgQuery)
+	if metadata.Org == "" {
+		webreg.MissingQuery(w, injectFromOrgQuery)
+		return metadata, test, fmt.Errorf("missing query %s", injectFromOrgQuery)
+	}
+	metadata.Repo = r.URL.Query().Get(injectFromRepoQuery)
+	if metadata.Repo == "" {
+		webreg.MissingQuery(w, injectFromRepoQuery)
+		return metadata, "", fmt.Errorf("missing query %s", injectFromRepoQuery)
+	}
+	metadata.Branch = r.URL.Query().Get(injectFromBranchQuery)
+	if metadata.Branch == "" {
+		webreg.MissingQuery(w, injectFromBranchQuery)
+		return metadata, test, fmt.Errorf("missing query %s", injectFromVariantQuery)
+	}
+	metadata.Variant = r.URL.Query().Get(injectFromVariantQuery)
+
+	test = r.URL.Query().Get(injectTestQuery)
+	if test == "" {
+		webreg.MissingQuery(w, injectTestQuery)
+		return metadata, test, fmt.Errorf("missing query %s", injectTestQuery)
+	}
+
+	return metadata, test, nil
+}
+
+func resolveConfigWithInjectedTest(configAgent agents.ConfigAgent, registryAgent agents.RegistryAgent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotImplemented)
+			_, _ = w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+		metadata, err := webreg.MetadataFromQuery(w, r)
+		if err != nil {
+			metrics.RecordError("invalid query", configresolverMetrics.ErrorRate)
+		}
+		logger := logrus.WithFields(api.LogFieldsFor(metadata))
+
+		config, err := configAgent.GetMatchingConfig(metadata)
+		if err != nil {
+			metrics.RecordError("config not found", configresolverMetrics.ErrorRate)
+			w.WriteHeader(http.StatusNotFound)
+			fmt.Fprintf(w, "failed to get config: %v", err)
+			logger.WithError(err).Warning("failed to get config")
+			return
+		}
+
+		injectFromMetadata, test, err := injectTestFromQuery(w, r)
+		if err != nil {
+			metrics.RecordError("invalid query", configresolverMetrics.ErrorRate)
```

Comment:
Do we still want to continue with the function if an error happens here?

Reply:
No. Wow, sloppy. Thanks!

Reply:
I just copied the code from the other method, so I added a commit that addresses this comment in all places.